src/invert.ts

```diff
@@ -0,0 +1,49 @@
+import { purry } from './purry';
+
+type Inverted<T extends object> = T[keyof T] extends PropertyKey
+  ? Record<T[keyof T], keyof T>
+  : never;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert(object)
+ * @example
+ *    R.invert({ a: "d", b: "e", c: "f" }) // => { d: "a", e: "b", f: "c" }
+ * @data_first
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(object: T): Inverted<T>;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert()(object)
+ * @example
+ *    R.pipe({ a: "d", b: "e", c: "f" }, R.invert()); // => { d: "a", e: "b", f: "c" }
+ * @data_last
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(): (object: T) => Inverted<T>;
+
+export function invert() {
+  return purry(_invert, arguments);
+}
+
+function _invert<T extends object>(object: T): Inverted<T> {
```

Comment:
Because of all the casts the implementation doesn't need to take the generic. You can type this a bit stricter and gain some help from TS. Now the values of this object are viable property keys and it might work without any error suppressions
```suggestion
function _invert(object: Readonly<Record<PropertyKey, PropertyKey>>): Record<PropertyKey, PropertyKey> {
```